src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs

```diff
@@ -0,0 +1,78 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Dapr.Client;
+
+namespace Microsoft.Extensions.Configuration.DaprSecretStore
+{
+    /// <summary>
+    /// A Dapr Secret Store based <see cref="ConfigurationProvider"/>.
+    /// </summary>
+    internal class DaprSecretStoreConfigurationProvider : ConfigurationProvider
+    {
+        private readonly string _store;
+
+        private readonly DaprSecretDescriptor[] _secrets;
+
+        private readonly DaprClient _client;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="DaprSecretStoreConfigurationProvider"/>.
+        /// </summary>
+        /// <param name="store">Dapr Secre Store name.</param>
+        /// <param name="secrets">The secrets to retrieve.</param>
+        /// <param name="client">Dapr client used to retrieve Secrets</param>
+        public DaprSecretStoreConfigurationProvider(string store, DaprSecretDescriptor[] secrets, DaprClient client)
+        {
+            if (store == null)
+            {
+                throw new ArgumentNullException(nameof(store));
+            }
+
+            if (secrets == null)
+            {
+                throw new ArgumentNullException(nameof(secrets));
+            }
+
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            _store = store;
+            _secrets = secrets;
+            _client = client;
+        }
+
+        public override void Load() => LoadAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+
+        private async Task LoadAsync()
+        {
+            var data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            foreach (var secret in _secrets)
+            {
+                var result = await _client.GetSecretAsync(_store, secret.SecretName, secret.Metadata);
```

Comment:
Needs a `ConfigureAwait(false)`

Reply:
Done in 3dcad42

Reply:
I thought `ConfigureAwait(false)` was only necessary in UI code.  What does it provide in this case?

Reply:
It's the good practice when writing libs, so if the lib ends up being used in a UI app it will not cause unwanted issues such as deadlocks.

Reply:
This is a library which can be used from UI code.

Reply:
based on that logic, every api in aspnetcore and other netcore libraries  should use ConfigureAwait false, but that's not the case, is there more granular reason for picking this.